stdlib/public/Darwin/Compression/Compression.swift

```diff
@@ -338,11 +338,11 @@ public class InputFilter<D: DataProtocol> {
     // Allocate result
     var result = Data(count: count)
 
-    try result.withUnsafeMutableBytes { (dst_ptr: UnsafeMutablePointer<UInt8>) in
+    try result.withUnsafeMutableBytes { (dst_ptr: UnsafeMutableRawBufferPointer) in
 
       // Write to result until full, or end reached
       _stream.dst_size = count
-      _stream.dst_ptr = dst_ptr
```

Comment:
`bindMemory` is not the right call here. @atrick is this one a good usecase for `assumingMemoryBound`?

Reply:
That's right. This is a good use of `assumingMemoryBound(to:)`. The C struct imported as `UnsafeMutablePointer<UInt8>` points into Foundation.Data's bytes, which don't have an element type. It's up to the user here to make sure that anything pointing into that Data object's bytes has `UInt8` type.

`bindMemory` is more appropriate when Data's bytes were used with some other type, and now you need to start using it with pointers of a different type.

Reply:
Fixed